Source/DafnyLanguageServer/DafnyLanguageServer.cs

```diff
@@ -19,21 +21,51 @@ public static class DafnyLanguageServer {
       }
     }
 
-    public static LanguageServerOptions WithDafnyLanguageServer(this LanguageServerOptions options, IConfiguration configuration) {
+    public static LanguageServerOptions WithDafnyLanguageServer(this LanguageServerOptions options,
+        IConfiguration configuration, Action killLanguageServer) {
```

Comment:
I think it makes more sense to name `killLanguageServer` like the event when it's raised rather than the action it's doing in a concrete implementation.

Reply:
I don't understand. I would name an event that is raised when a language server is killed `OnLanguageServerKilled`. But this argument is not an event, it's a function that when called will kill the language server. Shouldn't a function get the name of the action it performs?

Reply:
@camrein  I think that `killLanguageServer` is appropriate as well. This is not a callback about what to do when the language server is killed. It's actually the action that does kill the language server, and that has to be provided by the external object creating the language server.

Reply:
I was thinking about something like `OnShutDownRequested`. However, your arguments make sense including the decision to not make this parameter nullable.